src/main/java/seedu/financeit/goaltracker/GoalTracker.java

```diff
@@ -73,85 +137,123 @@ public static int getIncomeGoal() {
-            totalGoalList.addGoal(goalSetter);
+            try {
+                month = Month.of(Integer.parseInt(cmdPacket[3]));
+                incomeGoal = income;
+                goalSetter = new Goal(incomeGoal, "Income", month);
+                totalGoalList.addGoal(goalSetter);
+            } catch (DateTimeException e) {
+                UiManager.printWithStatusIcon(Constants.PrintType.ERROR_MESSAGE, "Invalid input. Please enter "
+                        + " a valid month");
+            }
         }
     }
 
-    private static void setExpenseGoal() {
-        backToMenu = false;
-        while (!backToMenu && exitTracker == false) {
-            System.out.println("Please enter your goals for your expenses.");
-            System.out.println("However if you wish to go back to Main Menu. Please enter <back>");
-            if (scanner.hasNextInt()) {
-                expenseGoal = scanner.nextInt();
-                System.out.println("You have set $" + expenseGoal + " as your Expense Goals");
-                goalSetter = new Goal(expenseGoal, "Expense");
-                totalGoalList.addGoal(goalSetter);
-                backToMenu = true;
-            } else if (scanner.hasNext()) {
-                input = scanner.next();
-                if (input.equals("back")) {
-                    backToMenu = true;
-                    main();
-                } else {
-                    UiManager.printWithStatusIcon(Constants.PrintType.SYS_MSG, "Invalid input. Please enter "
-                            + "<back> to go Main Menu or enter a number");
-                }
+    private static void displayGoal() {
+        try {
+            if (cmdPacket[1].equals("expense")) {
+                ledgerMonth = null;
+                handleExpenseGoal();
+            } else if (cmdPacket[1].equals("income")) {
+                ledgerMonth = null;
+                handleIncomeGoal();
             }
+        } catch (IndexOutOfBoundsException e) {
+            UiManager.printWithStatusIcon(Constants.PrintType.ERROR_MESSAGE, "Please enter either expense "
+            + "or income");
+            main();
         }
+    }
 
-        if (backToMenu == true && exitTracker == false) {
-            System.out.println("Do you want to exit Expense Goal Tracker? y/n ");
-            input = UiManager.handleInput();
-            cmdPacket = new InputParser().parseInput(input.toLowerCase());
-            if (cmdPacket.getCommandString().equals("y")) {
+    private static void setExpenseGoal() {
+        try {
+            isExist = false;
+            month = Month.of(Integer.parseInt(cmdPacket[3]));
+            expenseGoal = Integer.parseInt(cmdPacket[1]);
+            for (int i = 0; i < totalGoalList.getListSize(); i++) {
+                try {
+                    if (totalGoalList.getGoal().get(i).getExpenseMonth().equals(month)) {
+                        isExist = true;
+                        break;
```

Comment:
If goals in totalGoalList is changed to a hashMap then the goal for that month can be accessed with this.goals.get(month) in TotalGoalList.java instead of haping to use a for loop to search through all entries.